flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -2606,23 +2614,24 @@ struct GlobalOpConversion : public FIROpConversion<fir::GlobalOp> {
   }
 };
 
-// convert to LLVM IR dialect `load`
+/// `fir.load` --> `llvm.load`
 struct LoadOpConversion : public FIROpConversion<fir::LoadOp> {
   using FIROpConversion::FIROpConversion;
 
   mlir::LogicalResult
-  matchAndRewrite(fir::LoadOp load, OperandTy operands,
+  matchAndRewrite(fir::LoadOp load, OpAdaptor adaptor,
                   mlir::ConversionPatternRewriter &rewriter) const override {
     // fir.box is a special case because it is considered as an ssa values in
     // fir, but it is lowered as a pointer to a descriptor. So fir.ref<fir.box>
-    // and fir.box end up being the same llvm types and loading a fir.ref<box>
-    // is actually a no op in LLVM.
+    // and fir.box end up being the same llvm types and loading a
+    // fir.ref<fir.box> is actually a no op in LLVM.
     if (load.getType().isa<fir::BoxType>()) {
-      rewriter.replaceOp(load, operands[0]);
+      rewriter.replaceOp(load, adaptor.getOperands()[0]);
     } else {
-      auto ty = convertType(load.getType());
-      auto at = load->getAttrs();
-      rewriter.replaceOpWithNewOp<mlir::LLVM::LoadOp>(load, ty, operands, at);
+      mlir::Type ty = convertType(load.getType());
+      ArrayRef<NamedAttribute> at = load->getAttrs();
```

Comment:
`llvm::ArrayRef`

There are other classes named ArrayRef in flang, so it's good to use the namespace.